Cargo.toml

```diff
@@ -53,6 +53,11 @@ serde = { version = "1.0", features = ["derive"], default-features = false, opti
 winter_crypto = { version = "0.8", package = "winter-crypto", default-features = false }
 winter_math = { version = "0.8", package = "winter-math", default-features = false }
 winter_utils = { version = "0.8", package = "winter-utils", default-features = false }
+rand = {version = "0.8.5"}
```

Comment:
These will need to become:

```toml
rand = { version = "0.8.5", default-features = false, features = ["alloc"] }
num-complex = { version = "0.4.4", default-features = false }
num = { version = "0.4.1", default-features = false, features = ["alloc", "rand"] } # the "serde" feature also if that's needed
sha3 = { version = "0.10.8", default-features = false }
hex = { version = "0.4.3", default-features = false, features = ["alloc"] }
```

Otherwise, the no-std build of `miden-crypto` will not work properly (all of those crates enable their `std` features by default).

I don't think you need to enable their `std` features when `miden-crypto`'s `std` feature is enabled, except for `rand`. I haven't looked to see what degree of functionality is missing without the `std` feature in each of these crates (those with `alloc` features are likely no different, unless they depend on other libstd features other than heap-allocated structures).

Anyway, just a heads up!

Reply:
Thank you for the heads up, @bitwalker !

Reply:
Yes - let's update the dependencies as @bitwalker suggested (and make sure all dependencies are listed in the alphabetical order).

Also, it seems like the `hex` dependency may be needed only for tests? If so, it should go into `dev-dependencies`.

Reply:
Done!
I had to adapt the above with respect to `OsRng` and the `getrandom` feature flag